pkg/sidecar_mounter/sidecar_mounter.go

```diff
@@ -204,3 +212,102 @@ func logVolumeTotalSize(dirPath string) {
 		klog.Infof("total volume size of %v: %v bytes", dirPath, totalSize)
 	}
 }
+
+// collectMetrics collects metrics from the gcsfuse instance every 10 seconds.
+func collectMetrics(ctx context.Context, port, dirPath string) {
+	genNumber := 0
+	metricEndpoint := "http://localhost:" + port + "/metrics"
+	ticker := time.NewTicker(10 * time.Second)
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
+			newCtx, cancel := context.WithTimeout(ctx, 5*time.Second)
+			scrapeMetrics(newCtx, metricEndpoint, dirPath, genNumber)
+			deleteOutdatedFile(dirPath, genNumber-2)
```

Comment:
can you please elaborate why we are only deleting genNum-2? also best to put a comment in code

Reply:
Done.